tests/test-cases.js

```diff
@@ -2018,6 +2027,51 @@ describe('Unparse Tests', function() {
 
 
 var CUSTOM_TESTS = [
+	{
+		description: "Pause and resume works with headers and duplicate fields (Regression Test for Bug #985)",
+		expected: [[
+			["Column 1", "Column 2", "Column 3", "Column 4", "Column 5"],
+			["Column 1", "Column 2", "Column 3", "Column 4", "Column 5"],
+			["Column 1", "Column 2", "Column 3", "Column 4", "Column 5"],
+			["Column 1", "Column 2", "Column 3", "Column 4", "Column 5"]
+		], [
+			{ "Column 1": "R1C1", "Column 2": "", "Column 3": "R1C3", "Column 4": "", "Column 5": "" },
+			{ "Column 1": "R2C1", "Column 2": "", "Column 3": "", "Column 4": "", "Column 5": "" },
+			{ "Column 1": "R3C1", "Column 2": "", "Column 3": "", "Column 4": "R3C4", "Column 5": "" },
+			{ "Column 1": "R4C1", "Column 2": "", "Column 3": "", "Column 4": "", "Column 5": "" },
+		]],
+		run: function(callback) {
+			var inputString = [
+				"Column 1,Column 2,Column 3,Column 4,Column 5",
+				"R1C1,,R1C3,,",
+				"R2C1,,,,",
+				"R3C1,,,R3C4,",
+				"R4C1,,,,"
+			].join("\n");
+			var output = [];
+			var dataRows = [];
+			var headerResults = [];
+			Papa.parse(inputString, {
+				header: true,
+				step: function(results, parser) {
+					if (results)
+					{
+						headerResults.push(results.meta.fields);
```

Comment:
Why pusshing headers multiple times? Won't be better to just push when empty?

Reply:
I wanted to double-check that multiple passes returned the same headers. It looked like the original code could have updated the headers on each pass, which seemed problematic. Perhaps that question makes sense as a separate test, or do you think it isn't worth the check?